packages/components/src/Dialog/Dialog.tsx

```diff
@@ -60,79 +35,58 @@ const CloseButton = ({ className }: CloseButtonProps) => {
   );
 };
 
-/**
- * Search for a direct child that can act as title to improve accessibility.
- */
-const addTitleProps = (
-  children: ReactNode,
-  titleProps: HTMLAttributes<HTMLElement>
-) => {
-  const childs = Children.toArray(children);
-
-  const titleIndex = childs.findIndex(
-    child =>
-      isValidElement(child) &&
-      (child.type === Header || child.type === Headline)
-  );
-
-  // No child found that can act as title
-  if (titleIndex < 0) {
-    console.warn(
-      'No child in <Dialog> found that can act as title for accessibility. Please add a <Header> or <Headline> as direct child.'
-    );
-    return children;
-  }
-
-  // If we found a child, add the titleProps to it
-  const titleChild = cloneElement(
-    childs[titleIndex] as ReactElement<any>,
-    titleProps
-  );
-  childs.splice(titleIndex, 1, titleChild);
-
-  return childs;
-};
-
 // Props
 // ---------------
-export interface DialogChildProps {
-  close: DialogContextProps['close'];
-  titleProps: HTMLAttributes<HTMLElement>;
-}
-
-export interface DialogProps extends AriaDialogProps {
-  children?: ReactNode | ((props: DialogChildProps) => ReactNode);
+export interface DialogProps extends RAC.DialogProps {
   variant?: string;
   size?: string;
   closeButton?: boolean;
+  isNonModal?: boolean;
 }
 
 // Component
 // ---------------
-export const Dialog = ({
-  children,
+const _Dialog = ({
   variant,
   size,
   closeButton,
+  isNonModal,
   ...props
 }: DialogProps) => {
-  const ref = useRef(null);
-  const { close } = useDialogContext();
-  const { dialogProps, titleProps } = useDialog(props, ref);
-
   const classNames = useClassNames({ component: 'Dialog', variant, size });
-
+  const DialogWrapper = ({ children }: { children: ReactNode }) => {
+    if (isNonModal) return <>{children}</>;
```

Comment:
I don't know if this is right or not 
i am trying to do it like before , so Dialog can behave as a modal or as a normal dialog

Reply:
Maybe we don't need this anymore? what does the wrapper do?

Reply:
in the beginning i was confused about the Dialog and Modal 
what is normal is that dialog is just that box inside the modal (in most of libraries and based on my researches)
but what i saw in our old `Dialog` that it is a combination of(`modal` + `dialog`) and it can behave like the `modal` 
because there's `Modal` inside `DialogTrigger` component 
so i won't put Modal over the Dialog component where should it be ?
or should i do like what react-aria did 
```
<DialogTrigger>
  <Button>Sign up…</Button>
  <Modal>
    <Dialog>
      {({ close }) => (
        <form>
          <Heading slot="title">Sign up</Heading>
          <TextField autoFocus>
            <Label>First Name</Label>
            <Input />
          </TextField>
          <TextField>
            <Label>Last Name</Label>
            <Input />
          </TextField>
          <Button onPress={close}>
            Submit
          </Button>
        </form>
      )}
    </Dialog>
  </Modal>
</DialogTrigger>
```